### PR TITLE
MySQL: nullable information_schema.tables columns and add test-suite for MySQL8

### PR DESCRIPTION
**Pull request by vroldanbet**
A user reported in Discord the following error

```
unable to query DB stats: sql: Scan error on column index 0, name "TABLE_ROWS": converting NULL to uint64 is unsupported
````

Some `information_schema.tables` columns are nullable, including `table_rows` used for SpiceDB statistics reported in telemetry. Depending on the configuration, specially in MySQL8, those values may not be populate and causes SpiceDB server to fail to start.

This changes the code to use `NullInt64`, and also runs the entire test-suite with V8.

### CODE REVIEW

File: internal/datastore/mysql/datastore_test.go [hunk 1]
```diff
@@ -90,14 +90,38 @@ func createDatastoreTest(b testdatastore.RunningEngineForTest, tf datastoreTestF
...
+
+func TestMySQL5DatastoreWithTablePrefix(t *testing.T) {
+	t.Parallel()
+	b := testdatastore.RunMySQLForTestingWithOptions(t, testdatastore.MySQLTesterOptions{MigrateForNewDatastore: true, Prefix: "spicedb_"}, "")
+	dst := datastoreTester{b: b, t: t, prefix: "spicedb_"}
+	test.All(t, test.DatastoreTesterFunc(dst.createDatastore))
+}
+
+func additionalMySQLTests(t *testing.T, b testdatastore.RunningEngineForTest) {
+	reg := prometheus.NewRegistry()
```
Comment (josephschorr) [inline internal/datastore/mysql/datastore_test.go:122]: why do we need to register a new registry?

File: internal/datastore/mysql/datastore_test.go [hunk 2]
```diff
@@ -90,14 +90,38 @@ func createDatastoreTest(b testdatastore.RunningEngineForTest, tf datastoreTestF
...
+func TestMySQL5Datastore(t *testing.T) {
+	t.Parallel()
 	b := testdatastore.RunMySQLForTesting(t, "")
 	dst := datastoreTester{b: b, t: t}
 	test.All(t, test.DatastoreTesterFunc(dst.createDatastore))
+	additionalMySQLTests(t, b)
+}
+
+func TestMySQL8Datastore(t *testing.T) {
+	t.Parallel()
```
Comment (josephschorr) [inline internal/datastore/mysql/datastore_test.go:107]: I have concerns about parallel against running containers; are we sure this is stable?

Comment (vroldanbet) [reply internal/datastore/mysql/datastore_test.go:107]: I'll remove it

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**